### DISCUSSION TITLE
Add render_mode that allows materials with scissor/hash opacity to cast shadows that ignore their opacity

### DISCUSSION THREAD

**Discussion by Bordraw** [godotengine/godot-proposals#11273] [3D]
Something that I have run into recently and haven't been able to find a good workaround for: I am writing a shader that fades out walls between the camera and the player using some vertex magic. I am using alpha scissor for this for performance reasons but when the walls fade out, their shadows do too. It would be incredibly helpful if we had flags like 'shadow_draw_always' and 'shadow_draw_opaque' that let us customize this behavior so that we have fine control over how shadows are cast when using discard transparency (i.e. alpha scissor or hash).

**Comment by Calinou** [comment]
This can be addressed if you can query whether you're currently in the shadow pass: https://github.com/godotengine/godot-proposals/issues/4443
